lib/assembly/components.ex

```diff
@@ -0,0 +1,21 @@
+defmodule Assembly.Components do
+  @moduledoc """
+  Maintains all fo the inventory component logic
+  """
+
+  alias Assembly.Cache
+
+  def update_quantity_available(component_id, quantity),
+    do: Cache.put(component_id, quantity)
+
+  def quantity_available(component_id) do
+    case Cache.get(component_id) do
+      {:error, _reason} ->
+        # TODO: Handle errors
```

Comment:
As far as error handling, I'm thinking just having the cache `get/1` fall back to 0 would avoid a lot of this manual handling null cases.